DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/SwiftUI/DisplayList.swift

```diff
@@ -0,0 +1,265 @@
+    }
+
+    internal struct ViewUpdater {
+        /// The view cache maps an items from the display tree
+        /// to a platform view.
+        internal struct ViewCache {
+            internal struct Key: Hashable {
+                let id: Index.ID
+            }
+
+            let map: [ViewCache.Key: ViewInfo]
+        }
+
+        internal struct ViewInfo {
+            /// Original view's `.backgorundColor`.
+            let backgroundColor: CGColor?
+
+            /// Original view's `layer.borderColor`.
+            let borderColor: CGColor?
+
+            /// Original view's `layer.borderWidth`.
+            let borderWidth: CGFloat
+
+            /// Original view's `layer.cornerRadius`.
+            let cornerRadius: CGFloat
+
+            /// Original view's `.alpha` (between `0.0` and `1.0`).
+            let alpha: CGFloat
+
+            /// Original view's `.isHidden`.
+            let isHidden: Bool
+
+            /// Original view's `.intrinsicContentSize`.
+            let intrinsicContentSize: CGSize
+        }
+
+        /// The view cache holds references to UIKit instances.
+        ///
+        /// Children must be reflected on the main thread to inspect values
+        /// without retaining views or layers references.
+        let viewCache: ViewCache
+
+        /// The display list reflects the SwiftUI rendering tree.
+        ///
+        /// Reflection is made lazely and can be performed on a
+        /// background thread.
+        let lastList: DisplayList.Lazy
+    }
+
+    internal struct Index {
+        internal struct ID: Hashable {
+            let identity: Identity
+        }
+    }
+
+    internal enum Effect {
+        case identify
+        case clip(SwiftUI.Path, SwiftUI.FillStyle)
+        case unknown
+    }
+
+    /// The content of a leaf item in the display list.
+    internal struct Content {
+        internal enum Value {
+            case shape(SwiftUI.Path, ResolvedPaint, SwiftUI.FillStyle)
+            case text(StyledTextContentView, CGSize)
+            case platformView
+            case color(Color._Resolved)
+            case unknown
+        }
+
+        let seed: Seed
+        let value: Value
+    }
+
+    /// An item is an element of the displayed tree in `SwiftUI`.
+    ///
+    /// The item has a unique identity, a frame in the current branch, and a value.
+    /// A value can be:
+    /// - An effect (node): which applied an effect on a sub-tree.
+    /// - A content (leaf): The content of the displayed item.
+    internal struct Item {
+        internal enum Value {
+            case effect(Effect, DisplayList)
+            case content(Content)
+            case unknown
+        }
+
+        let identity: Identity
+        let frame: CGRect
+        let value: Value
+    }
+
+    let items: [Item]
+}
+
+@available(iOS 13.0, *)
+extension DisplayList: Reflection {
+    init(_ mirror: Mirror) throws {
+        items = try mirror.descendant(path: "items")
```

Comment:
It's already safely cast by the `mirror.descendant(path:)`, it infers the return type from `self.items`. If the type mismatch, or fails to reflect, it will throw an error.